Source/Dafny/AST/DafnyAst.cs

```diff
@@ -1061,6 +1065,30 @@ public enum AutoInitInfo { MaybeEmpty, Nonempty, CompilableValue }
       }
     }
 
+    // Returns true if it's possible to check that a value has this type at run-time.
+    public bool IsCompilable() {
```

Comment:
The problem will only arise if we add more types that cannot be checked at run-time, meaning they carry ghost constraints. This can be the case only when there is a subset type declaration, or when the type has not been inferred yet. That's the only two possible case where we cannot say for sure that the type is compilable.
Compilability being used only in the case of type conversion (i.e. checking for a constraint), it is not a property `newtype`s need because their constraint does not need to be checked at run-time.

Reply:
Yes, I understand the above, but still shouldn't the type specify whether it's possible to check that a value has this type at run-time?